rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +924,45 @@ void saveFederationElection_shouldSaveInStorage() {
         assertArrayEquals(federationElectionSerializedWithVote, actualAbiCallElectionSerialized);
     }
 
+    @Test
+    void saveProposedFederation_whenProposedFederationIsNotSet_shouldDoNothing() {
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.lovell700().forBlock(0L);
+        StorageAccessor bridgeStorageAccessor = mock(BridgeStorageAccessorImpl.class);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+
+        federationStorageProvider.save(networkParameters, activations);
+
+        verify(bridgeStorageAccessor, never()).saveToRepository(any(), any(), any());
+    }
+
+    @Test
+    void saveProposedFederation_preRSKIP419_whenProposedFederationIsSet_shouldDoNothing() {
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.arrowhead631().forBlock(0L);
+        StorageAccessor bridgeStorageAccessor = mock(BridgeStorageAccessorImpl.class);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        Federation proposedFederation = new P2shErpFederationBuilder().build();
+
+        federationStorageProvider.setProposedFederation(proposedFederation);
+        federationStorageProvider.save(networkParameters, activations);
+
+        verify(bridgeStorageAccessor, never()).saveToRepository(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), proposedFederation.getFormatVersion(), BridgeSerializationUtils::serializeInteger);
+        verify(bridgeStorageAccessor, never()).saveToRepository(PROPOSED_FEDERATION.getKey(), proposedFederation, BridgeSerializationUtils::serializeFederation);
+    }
+
+    @Test
+    void saveProposedFederation_whenProposedFederationIsSet_shouldSave() {
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.lovell700().forBlock(0L);
+        StorageAccessor bridgeStorageAccessor = mock(BridgeStorageAccessorImpl.class);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        Federation proposedFederation = new P2shErpFederationBuilder().build();
+
+        federationStorageProvider.setProposedFederation(proposedFederation);
+        federationStorageProvider.save(networkParameters, activations);
+
+        verify(bridgeStorageAccessor).saveToRepository(eq(PROPOSED_FEDERATION_FORMAT_VERSION.getKey()), eq(proposedFederation.getFormatVersion()), any());
+        verify(bridgeStorageAccessor).saveToRepository(eq(PROPOSED_FEDERATION.getKey()), eq(proposedFederation), any());
```

Comment:
If we use InMemoryStorage, we could also read the stored value under proposed federation keys and assert that what is stored is what is expected. That would make the test really solid